runtime/tchannel/server.go

```diff
@@ -0,0 +1,216 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package tchannel
+
+import (
+	"bytes"
+	"log"
+	"strings"
+	"sync"
+
+	tchan "github.com/uber/tchannel-go"
+	"github.com/uber/tchannel-go/thrift"
+
+	"go.uber.org/thriftrw/protocol"
+	"go.uber.org/thriftrw/wire"
+	"golang.org/x/net/context"
+)
+
+type handler struct {
+	server         TChanServer
+	postResponseCB PostResponseCB
+}
+
+// Server handles incoming TChannel calls and forwards them to the matching TChanServer.
+type Server struct {
+	sync.RWMutex
+	ch       tchan.Registrar
+	log      tchan.Logger
+	handlers map[string]handler
+	ctxFn    func(ctx context.Context, method string, headers map[string]string) thrift.Context
+}
+
+// NewServer returns a server that can serve thrift services over TChannel.
+func NewServer(registrar tchan.Registrar) *Server {
+	server := &Server{
+		ch:       registrar,
+		log:      registrar.Logger(),
```

Comment:
https://godoc.org/github.com/uber/tchannel-go#Logger

The logger interface is not the same as zap.Logger, we need to pass zanzibar logger down directly instead of using it from `registrar`.